Tasks/FtpUpload/task.json

```diff
@@ -0,0 +1,97 @@
+{
+  "id": "6f8c69a5-b023-428e-a125-fccf4efcb929",
+  "name": "FtpUpload",
+  "friendlyName": "Ftp Upload",
+  "description": "Ftp Upload",
+  "author": "Microsoft Corporation",
+  "helpMarkDown": "Upload files to an FTP server using either FTP, or secure FTPS",
+  "category": "Utility",
+  "visibility": [
+    "Build",
+    "Release"
+  ],
+  "demands": [],
+  "version": {
+    "Major": "1",
+    "Minor": "0",
+    "Patch": "0"
+  },
+  "instanceNameFormat": "FTP Upload: $(rootFolder)",
+  "groups": [
+        {
+          "name": "advanced",
+          "displayName": "Advanced",
+          "isExpanded": false
+        }
+    ],
+  "inputs": [
+    {
+      "name": "serverEndpoint",
+      "type": "connectedService:Generic",
+      "label": "FTP Service Endpoint",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Select the service endpoint for your FTP server.  To create one, click the Manage link and create a new Generic Service Endpoint, enter the FTP server URL for the server URL, e.g. ftps://server.example.com, and required credentials."
+    },
+    {
+      "name": "rootFolder",
+      "type": "filePath",
+      "label": "Source Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": ""
+    },
+    {
+      "name": "filePatterns",
+      "type": "multiLine",
+      "label": "File patterns",
+      "defaultValue": "*",
+      "required": true,
+      "helpMarkDown": "File paths or patterns of the files to upload.  Supports multiple lines of minimatch patterns.  [More Information](https://go.microsoft.com/fwlink/?LinkId=800269)",
+      "properties": {
+          "resizable": "true",
+          "rows" : "4"
+      }
+    },
+    {
+      "name": "remotePath",
+      "type": "string",
+      "label": "Remote directory to upload files",
+      "defaultValue": "/upload/$(Build.BuildId)/",
+      "required": true,
+      "helpMarkDown": ""
+    },
+    {
+        "name": "clean",
+        "type": "boolean",
+        "label": "Clean Remote Folder",
+        "defaultValue": "false",
+        "required": true,
+        "helpMarkDown": "Delete the remote directory location before uploading",
+        "groupName": "advanced"
+    },
+    {
+        "name": "overwrite",
+        "type": "boolean",
+        "label": "Overwrite",
+        "defaultValue": "true",
+        "required": true,
+        "helpMarkDown": "Overwrite existing files in the remote location",
+        "groupName": "advanced"
+    },
+    {
+        "name": "flatten",
+        "type": "boolean",
+        "label": "Flatten Paths",
+        "defaultValue": "false",
+        "required": true,
+        "helpMarkDown": "Flatten the paths at the remote location",
```

Comment:
For additional clarity, let's explain this further.  Something like:
"If selected, files are uploaded directly into the remote directory without creating additional directories. Otherwise, the local directory structure is recreated where files are uploaded in the remote directory."